In [75]:
#Packages
#---------------------------#

# Datamanipulation Packages
import pandas as pd
import numpy as np


#DataVisualization Packages
import matplotlib.pyplot as plt
import seaborn as sns

# Modeling Packages
from xgboost import XGBClassifier
from xgboost import XGBRegressor
from xgboost import plot_importance

from sklearn.linear_model import  LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier

#Transformer
from sklearn.preprocessing import LabelEncoder


# ignore warnings   
import warnings
warnings.filterwarnings('ignore')

# Display all columns in dataframes
pd.set_option('display.max_columns', None)


In [89]:
#Load Data
df = pd.read_csv("Data/email.csv")
df.head()

Category                                            Message
0      ham  Go until jurong point, crazy.. Available only ...
1      ham                      Ok lar... Joking wif u oni...
2     spam  Free entry in 2 a wkly comp to win FA Cup fina...
3      ham  U dun say so early hor... U c already then say...
4      ham  Nah I don't think he goes to usf, he lives aro...

In [77]:
#Data Inspection

#Check Columns
print("The Data Info:")
print("------------")
df.info()

print("\n")


#CHeck Null columns
print("Null COlumns ")
print("-------------")
print(df.isna().sum())
print("\n")


#Check Duplicates
print("Duplicates")
print("----------")
print(df.duplicated().sum())
print("\n")


print("Count values in the category column")
print("-------------------------------------")
print(df['Category'].value_counts())






The Data Info:
------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5573 entries, 0 to 5572
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Category  5573 non-null   object
 1   Message   5573 non-null   object
dtypes: object(2)
memory usage: 87.2+ KB


Null COlumns 
-------------
Category    0
Message     0
dtype: int64


Duplicates
----------
415


Count values in the category column
-------------------------------------
Category
ham               4825
spam               747
{"mode":"full"       1
Name: count, dtype: int64


#### Observations

1. There are 415 Duplicactes --------------------------------------------------------> Remove all of them
2. There are 2 columns all of type object--------------------------------------------> COnvert category column into integer datatype using label encoder or  maping or lambda function
3. There are 4825 values for ham and 747 values for spam; 1 value for {"mode":"full" --> Drop this record

### Data Cleaning

In [88]:
#Drop {"mode":"full" reocrd

df1 = df[df["Category"] != '{"mode":"full"']

#Transform Category column to numeric datatype
#df1['Category']=df1['Category'].apply(lambda x:1 if x=='spam' else 0) # Does same thing as the Label ENcoder Method Below
encoder = LabelEncoder()
df1['Category'] = encoder.fit_transform(df1['Category'])

#Drop Duplicates
df1.drop_duplicates(keep='first', inplace=True)


